src/core/lib/surface/call.cc

```diff
@@ -912,6 +800,10 @@ grpc_error_handle FilterStackCall::Create(grpc_call_create_args* args,
     }
   }
 
+  if (args->send_deadline != Timestamp::InfFuture()) {
+    call->UpdateDeadline(args->send_deadline);
```

Comment:
Shouldn't this be done only under the same conditions that we were doing it previously?

https://github.com/grpc/grpc/blob/7f3bf40edb3888a539b9ad08ca8f2e32b7377f7b/src/core/client_channel/client_channel_filter.cc#L1233

Reply:
Removing that knob... it's very hard to support if we don't have a deadline filter, and it was added to try and justify the need for the deadline filter. If anyone's using it, they can set an infinite deadline too.